walkthroughs/howto-mutual-tls-file-provided-by-acm/infrastructure/ecs-service.yaml

```diff
@@ -0,0 +1,412 @@
+        MinimumHealthyPercent: 100
+      DesiredCount: 1
+      LaunchType: FARGATE
+      ServiceRegistries:
+        - RegistryArn:
+            'Fn::GetAtt': ColorTellerServiceDiscoveryRecord.Arn
+      NetworkConfiguration:
+        AwsvpcConfiguration:
+          AssignPublicIp: DISABLED
+          SecurityGroups:
+            - 'Fn::ImportValue': !Sub "${EnvironmentName}:ECSServiceSecurityGroup"
+          Subnets: 
+            - 'Fn::ImportValue': !Sub "${EnvironmentName}:PrivateSubnet1"
+            - 'Fn::ImportValue': !Sub "${EnvironmentName}:PrivateSubnet2"
+      TaskDefinition: { Ref: ColorTellerTaskDefinition }
+
+  GatewayServiceDiscoveryRecord:
+    Type: 'AWS::ServiceDiscovery::Service'
+    Properties:
+      Name: "gateway"
+      DnsConfig:
+        NamespaceId:
+          'Fn::ImportValue': !Sub "${EnvironmentName}:ECSServiceDiscoveryNamespace"
+        DnsRecords:
+          - Type: A
+            TTL: 300
+      HealthCheckCustomConfig:
+        FailureThreshold: 1
+
+  GatewayTaskDefinition:
+    Type: AWS::ECS::TaskDefinition
+    Properties:
+      RequiresCompatibilities:
+        - 'FARGATE'
+      Family: 'gateway'
+      NetworkMode: 'awsvpc'
+      Cpu: 256
+      Memory: 512
+      TaskRoleArn:
+        'Fn::ImportValue': !Sub "${EnvironmentName}:TaskIamRoleArn"
+      ExecutionRoleArn:
+        'Fn::ImportValue': !Sub "${EnvironmentName}:TaskExecutionIamRoleArn"
+      ContainerDefinitions:
+        - Name: envoy
+          Image: !Sub '${AWS::AccountId}.dkr.ecr.${AWS::Region}.amazonaws.com/${EnvoyImageName}'
+          Essential: true
+          User: '1337'
+          Ulimits:
+            - Name: "nofile"
+              HardLimit: 15000
+              SoftLimit: 15000
+          PortMappings:
+            - ContainerPort: 9080
+              Protocol: 'tcp'
+            - ContainerPort: 9901
+              Protocol: 'tcp'
+          HealthCheck:
+            Command:
+              - 'CMD-SHELL'
+              - 'curl -s http://localhost:9901/server_info | grep state | grep -q LIVE'
+            Interval: 5
+            Timeout: 2
+            Retries: 3
+          LogConfiguration:
+            LogDriver: 'awslogs'
+            Options:
+              awslogs-group:
+                Fn::ImportValue: !Sub "${EnvironmentName}:ECSServiceLogGroup"
+              awslogs-region: !Ref AWS::Region
+              awslogs-stream-prefix: 'gateway'
+          Environment:
+            - Name: 'APPMESH_VIRTUAL_NODE_NAME'
+              Value: !Sub 'mesh/${AppMeshMeshName}/virtualGateway/gateway-vgw'
+            - Name: AWS_REGION
+              Value: !Ref 'AWS::Region'
+            - Name: ENVOY_LOG_LEVEL
+              Value: warning
+            
+          # Set secret environment variable for the container
+          Secrets:
+            - Name: CertSecret
+              ValueFrom: {'Fn::ImportValue': !Sub "${EnvironmentName}:SecretCertArn"}
+
+  GatewayService:
+    Type: 'AWS::ECS::Service'
```

Comment:
you need a `DependsOn: PublicLoadBalancer` here. or else you can get an exception: "Target group 'WebTargetGroup' doesn't have an associated load balancer"

Reply:
I added the `DependsOn` to the WebTargetGroup directly
https://github.com/aws/aws-app-mesh-examples/blob/b32767f5c4b0a91912fe9b9e3486e01b6b39fc6e/walkthroughs/howto-mutual-tls-file-provided-by-acm/infrastructure/ecs-service.yaml#L255-L256